cmd/aries-agent-lambda/aries.go

```diff
@@ -0,0 +1,387 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+Copyright Boran Car <boran.car@gmail.com>. All Rights Reserved.
+Copyright Christian Nuss <christian@scaffold.ly>, Founder, Scaffoldly LLC. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"crypto/subtle"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/hyperledger/aries-framework-go/pkg/controller"
+	"github.com/hyperledger/aries-framework-go/pkg/controller/command"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/messaging/msghandler"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/transport"
+	wsinternal "github.com/hyperledger/aries-framework-go/cmd/aries-agent-lambda/transport/apigw_ws"
+	arieshttp "github.com/hyperledger/aries-framework-go/pkg/didcomm/transport/http"
+	httpinternal "github.com/hyperledger/aries-framework-go/cmd/aries-agent-lambda/transport/apigw_http"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries"
+	ariescontext "github.com/hyperledger/aries-framework-go/pkg/framework/context"
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
+	"github.com/hyperledger/aries-framework-go/pkg/vdr/httpbinding"
+	"github.com/hyperledger/aries-framework-go/cmd/aries-agent-lambda/storage/dynamodb"
+)
+
+type dbParam struct {
+	dbType string
+	// prefix  string
+	// timeout uint64
+}
+
+type agentParameters struct {
+	// server                                         server
+	host, defaultLabel, transportReturnRoute       string
+	tlsCertFile, tlsKeyFile                        string
+	token, keyType, keyAgreementType               string
+	webhookURLs, httpResolvers, outboundTransports []string
+	inboundHostInternals, inboundHostExternals     []string
+	contextProviderURLs, mediaTypeProfiles         []string
+	autoAccept                                     bool
+	msgHandler                                     command.MessageHandler
+	dbParam                                        *dbParam
+	autoExecuteRFC0593                             bool
+}
+
+const (
+	httpProtocol      = "http"
+	websocketProtocol = "ws"
+	defaultWebhookUrl = "https://sly.slygo.scaffoldly.dev/aries-webhooks/api/v1/webhooks"
```

Comment:
We've noticed we needed a webhook, otherwise we wouldn't get didcomm working. Can you point us to an easy way of avoiding the need for a webhook?

Reply:
@rolsonquadras @fqutishat @sudeshrshetty

Reply:
since you are using aries REST controller, you need webhook to get notification of incoming protocol messages during various protocol execution.



If you don't want to host external webhooks then you can use `controller.WithNotifier()` instead of `controller.WithWebhookURLs()` to provide your own custom notifier.

Reply:
since you are using aries REST controller, you need webhook to get notification of incoming protocol messages during various protocol execution.



If you don't want to host external webhooks then you can use `controller.WithNotifier()` instead of `controller.WithWebhookURLs()` to provide your own custom notifier.